src/main/java/org/opentripplanner/transit/model/network/StopPattern.java

```diff
@@ -323,6 +325,20 @@ public StopPatternBuilder cancelStops(List<Integer> cancelledStopIndices) {
       return this;
     }
 
+    /**
+     * Replace the stop {@code old} in the stop pattern with ${code newStop}.
+     */
+    public StopPatternBuilder replaceStop(FeedScopedId old, StopLocation newStop) {
```

Comment:
This breaks encapsulation and is dangerous. I would prefer that you create a new StopPattern instead. I know that the model i broken and that maintaining the integrity is problematic. Can you take a look to see how much work it is to change it? And then we can discuss what to do?

Reply:
Sure, I can do that, but are you aware that this is operating on a the `StopPatternBuilder` rather than on the `StopPattern` itself?

I believe that I already create a new stop and trip patterns and mutate those instead of the originals.

The code that does it looks like this: https://github.com/opentripplanner/OpenTripPlanner/blob/17c513cbc10b00388d53d123d5c66f5247aa75bd/src/ext/java/org/opentripplanner/ext/stopconsolidation/StopConsolidationModule.java#L68-L70

Does this already sastisfy your requirements (no in-place updates) or am I missing something?

Reply:
Sorry, I missed the fact that this is in the Bulder. GitHub is hiding to much context.